### PR TITLE
Add ReScript language server configuration

### PR DESCRIPTION
**Pull request by amiralies**

### CODE REVIEW

**Comment by mjlbach** [comment]
Don't modify Config.md directly please, it's supposed to be auto-generated.

File: lua/lspconfig/rescriptls.lua [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    root_dir = lspconfig.util.root_pattern('bsconfig.json'),
+    settings = {}
+  },
+  docs = {
+    description = [[
+https://github.com/rescript-lang/rescript-vscode
+ReScript language server
+Clone the repo and compile the language server,
+Set `cmd` to `cmd = {'node', '<path_to_repo>/server/out/server.js', '--stdio'}`
+If you're using vim-rescript you can also use the bundled server.js inside vim-rescript repo,
```
Comment (mjlbach) [inline lua/lspconfig/rescriptls.lua:19]: inside *the*

**Comment by amiralies** [comment]
> Don't modify Config.md directly please, it's supposed to be auto-generated.

My bad, done.

**Comment by amiralies** [comment]
Done.

**Comment by rocinant3** [comment]
Can anyone share vim configuration for working with Rescript on a pure LSP? (At the moment, it turned out to get COC support, but LSP is needed)

**Pull request merged**